src/root/Project/Glance/Arrivals.js

```diff
@@ -0,0 +1,229 @@
+import {Observable as $} from 'rx'
+import isolate from '@cycle/isolate'
+import {complement, isEmpty, filter, any, split, toLower, compose,
+        map, prop, props, allPass, isNil, anyPass, join, useWith,
+        propEq, head, unless, always} from 'ramda'
+
+import {div, img, span} from 'cycle-snabbdom'
+import {combineDOMsToDiv, formatTime, log} from 'util'
+
+import {iconSrc} from 'helpers'
+
+import {
+  List,
+  ListItem,
+  ListItemClickable,
+  InputControl,
+  LargeCard,
+  TitledCard,
+} from 'components/sdm'
+
+import {
+  ShiftContentExtra,
+} from 'components/shift'
+
+import {
+  Arrivals,
+} from 'components/remote'
+
+import AssignmentsFetcher from './AssignmentsFetcher'
+
+require('./arrivals.scss')
+
+const SearchBox = sources => {
+  const focus$ = sources.DOM.select('.arrivals-search').observable
+    .distinctUntilChanged()
```

Comment:
is it worth fixing even? I could just take out the distinct, as this seems to work fine anyway